hack/e2e.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash -e
+#
+# This script is intended to be run from the CI pipeline as follows:
+#
+#   # Define secrets in the envionment
+#   source hack/e2e.sh
+#   aro_ci_setup
+#   make e2e
+
+get_random_location() {
+    array[0]="australiasoutheast"
+    array[1]="eastus"
+    array[2]="westeurope"
+
+    size=${#array[@]}
+    index=$(($RANDOM % $size))
+    echo ${array[$index]}
+}
+
+aro_ci_setup() {
+    echo "====== starting dev RP =========================="
+    go build ./cmd/aro
+    ./aro rp &
+    trap 'return_id=$?; aro_ci_teardown_handler; exit $return_id' EXIT
+    while [[ "$(curl -s -o /dev/null -w '%{http_code}' localhost:8443)" == "000" ]]; do
+        sleep 2
+    done
+
+    if $CLUSTER_CREATE; then
+        echo "====== setup subnets =========================="
+        for subnet in "$CLUSTER-master" "$CLUSTER-worker"; do
+            az network vnet subnet create \
```

Comment:
Should we just add those into `az` as helpers and eliminate them from here? @jim-minter ?

Reply:
I think this is a valid architectural topic to bring up with MSFT - please could you?  My current view is that on balance I don't think it should be implemented in `az`, but it's not a super strong feeling and I'm open to be persuaded.